lib/baseTest.py

```diff
@@ -132,21 +136,39 @@ def remove_test_file(self):
         if hasattr(self, "test_url_id"):
             self.target_helper.delete_target(self.test_url_id)
 
-    def load_configs(self):
+    def _platform_dep_settings_handler(self, config_value):
+        if self.current_platform_name in config_value:
+            if self.current_platform_ver in config_value[self.current_platform_name]:
+                platform_dep_variables = copy.deepcopy(config_value[self.current_platform_name][self.current_platform_ver])
+            else:
+                platform_dep_variables = copy.deepcopy(config_value[self.current_platform_name][''])
+            config_value.update(platform_dep_variables)
+            return platform_dep_variables
+        return {}
+
+    def set_configs(self, config_variable_name, config_value):
```

Comment:
I will suggest to add a class to record the accept config name
```python
class ConfigName():
    EXEC = 'exec_config'
    INDEX = 'index_config'
    # ...
```

And then check the config name before overwrite the original variable
```python
ACCEPT_CONFIG_LIST = [ConfigName.EXEC, ConfigName.INDEX, ...]
# ...
if config_variable_name not in ACCEPT_CONFIG_LIST:
    raise Exception('Do not accept overwrite the {config_name} : {config_value}'.format(config_name= config_variable_name, config_value=config_value))
```

Also, the cases can import `ConfigName` class to overwrite the settings easily
```python
self.set_configs(ConfigName.INDEX, {"platform-dep-settings": {"win32": {"": {"compare-threshold": 0.003}}}})
```